src/jit/morph.cpp

```diff
@@ -12503,31 +12503,44 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
 
             /* See if we can fold GT_ADD nodes. */
 
-            if (oper == GT_ADD)
+            if (oper == GT_ADD || oper == GT_OR)
```

Comment:
Why not add `GT_MUL`, `GT_AND` and `GT_XOR` as well?

Reply:
Also, the comment above should be changed - probably to:
```
// See if we can fold constants for commutative operators.
```

Reply:
@mikedn good point, I was only keeping in mind the Enum and OR case (it's quite popular)